### PR TITLE
Renesas RX TSIP ECDSA support

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Support for Renesas RX TSIP with ECDSA and Crypto Callbacks.

* Fix building ECC with NO_ASN (`./configure --enable-cryptonly --disable-rsa --disable-asn --disable-examples`). Adds support for raw R+S signature.
* Fix for building `WOLFSSL_RENESAS_TSIP_CRYPTONLY` and `NO_WOLFSSL_RENESAS_TSIP_CRYPT_HASH`.
* Revert built-in `wc_GenerateSeed` support for RX TSIP removed in #6851.
* Updated example key data for private key for client authentication

# Testing

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by dgarske** [comment]
@kojo1 or @miyazakh can you review these changes and possibly test them out with e2Studio. I want to make sure I didn't break anything.

**Comment by dgarske** [comment]
Retest this please

**Comment by dgarske** [comment]
Retest this please

**Review by miyazakh — DISMISSED** [review]

**Comment by miyazakh** [comment]
Hi @dgarske 
Confirmed that tests including client/server example are passed.

**Comment by miyazakh** [comment]
Retest this please

**Comment by miyazakh** [comment]
Confirmed that tests including client/server example are passed on the latest change

**Review by miyazakh — DISMISSED** [review]

**Comment by dgarske** [comment]
@miyazakh if you are happy with the PR please indicate approved. I'd like to get this PR merged in soon.

File: IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/user_settings.h [hunk 1]
```diff
@@ -245,8 +245,7 @@
      */
     #define WOLFSSL_RENESAS_TSIP_TLS
 
-    #if !defined(NO_RENESAS_TSIP_CRYPT)
-        #define WOLFSSL_RENESAS_TSIP_CRYPTONLY
+	#if !defined(NO_RENESAS_TSIP_CRYPT)
```
Comment (SparkiDev) [inline IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/user_settings.h:248]: Doesn't look aligned with #endif

File: wolfcrypt/src/port/Renesas/renesas_tsip_util.c [hunk 2]
```diff
@@ -4042,7 +4006,111 @@ WOLFSSL_LOCAL int tsip_SignEcdsa(wc_CryptoInfo* info, TsipUserCtx* tuc)
...
+        hashData.data_type  = tuc->keyflgs_crypt.bits.message_type;
+        sigData.pdata       = (uint8_t*)info->pk.eccverify.sig;
+        sigData.data_length = info->pk.eccverify.siglen;
+
+        if ((ret = tsip_hw_lock()) == 0) {
+            switch (tuc->wrappedKeyType) {
+            #if !defined(NO_ECC256)
+                case TSIP_KEY_TYPE_ECDSAP256:
+                    /* zero pad or truncate */
+                    hashData.data_length = tsip_HashPad(32, hash,
```
Comment (SparkiDev) [inline wolfcrypt/src/port/Renesas/renesas_tsip_util.c:4066]: Replace 32 with a #define.

File: wolfcrypt/src/port/Renesas/renesas_tsip_util.c [hunk 3]
```diff
@@ -4042,7 +4006,111 @@ WOLFSSL_LOCAL int tsip_SignEcdsa(wc_CryptoInfo* info, TsipUserCtx* tuc)
...
+                        ret = WC_HW_E;
+                    }
+                    break;
+                    break;
+            #endif
+
+            #if defined(HAVE_ECC384)
+                case TSIP_KEY_TYPE_ECDSAP384:
+                    /* zero pad or truncate */
+                    hashData.data_length = tsip_HashPad(48, hash,
```
Comment (SparkiDev) [inline wolfcrypt/src/port/Renesas/renesas_tsip_util.c:4084]: Replace 48 with a #define.

File: wolfcrypt/src/port/Renesas/renesas_tsip_util.c [hunk 4]
```diff
@@ -4042,7 +4006,111 @@ WOLFSSL_LOCAL int tsip_SignEcdsa(wc_CryptoInfo* info, TsipUserCtx* tuc)
...
+ *   0 on success, CRYPTOCB_UNAVAILABLE on unsupported key type specified.
+ */
+WOLFSSL_LOCAL int tsip_VerifyEcdsa(wc_CryptoInfo* info, TsipUserCtx* tuc)
+{
+    int ret = 0;
+    e_tsip_err_t    err = TSIP_SUCCESS;
+    tsip_ecdsa_byte_data_t hashData, sigData;
+    /* hard coding largest digest size, since WC_MAX_DIGEST_SZ could be 32
+     * if using SHA2-256 with ECDSA SECP384R1 */
+    uint8_t hash[48];
```
Comment (SparkiDev) [inline wolfcrypt/src/port/Renesas/renesas_tsip_util.c:4040]: Replace 48 with a #define.

File: wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h [hunk 5]
```diff
@@ -121,26 +132,33 @@ typedef struct MsgBag
...
-    uint8_t aes256_key_set:1;
-    uint8_t aes128_key_set:1;
-    uint8_t rsapri2048_key_set:1;
-    uint8_t rsapub2048_key_set:1;
-    uint8_t rsapri1024_key_set:1;
-    uint8_t rsapub1024_key_set:1;
-    uint8_t message_type:1;/*message 0, hashed 1*/
- };
+/* flags Crypt Only */
+struct tsip_keyflgs_cryt {
```
Comment (SparkiDev) [inline wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h:136]: cryt???

File: wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h [hunk 6]
```diff
@@ -401,22 +416,29 @@ WOLFSSL_API void tsip_inform_user_keys(
 /*----------------------------------------------------*/
 /*   internal use functions                           */
 /*----------------------------------------------------*/
-WOLFSSL_LOCAL int tsip_SignRsaPkcs(wc_CryptoInfo* info, TsipUserCtx* tuc);
-
+#ifdef HAVE_PK_CALLBACKS
 WOLFSSL_LOCAL int tsip_VerifyRsaPkcsCb(
-                        WOLFSSL* ssl,
+                        struct WOLFSSL* ssl,
```
Comment (SparkiDev) [inline wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h:421]: This needs 'struct'???

File: wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h [hunk 7]
```diff
@@ -87,16 +95,19 @@ enum {
 };
 
 typedef enum {
-    TSIP_KEY_TYPE_RSA2048      = 0,
-    TSIP_KEY_TYPE_RSA4096      = 1,
-    TSIP_KEY_TYPE_ECDSAP256    = 2,
     #ifdef WOLFSSL_RENESAS_TSIP_CRYPTONLY
-    TSIP_KEY_TYPE_RSA1024      = 3,
+    TSIP_KEY_TYPE_RSA1024      = 1, /* TSIP_RSA1024 */
```
Comment (SparkiDev) [inline wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h:99]: Can we not assign one to the other?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h (hunk 6 above)
Comment (dgarske) [reply wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h:421]: Opaque pointer, so yes.

File: wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h (hunk 7 above)
Comment (dgarske) [reply wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h:99]: The headers are a mess. I spent about an hour trying to get one to use the other (either direction) without success.

**Review by SparkiDev — APPROVED** [review]

**Comment by SparkiDev** [comment]
retest this please

**Pull request merged**